neo4j/internal/pool/pool.go

```diff
@@ -145,7 +145,7 @@ func (p *Pool) CleanUp() {
 	}
 }
 
-func (p *Pool) tryBorrow(serverName string) (db.Connection, error) {
+func (p *Pool) tryBorrow(serverName string, boltLogger log.BoltLogger) (db.Connection, error) {
```

Comment:
Maybe I'm missing where you are doing it but don't you need to reset to tracer somewhere, like in pool.Return? Otherwise a session that enables tracing will set the tracer on the connection and it will be there until it is timed out from the pool.

Reply:
I could indeed null out the logger when the connection is returned (note this that won't have any observable effect afaik)